cmd/swarmctl/service/flagparser/update.go

```diff
@@ -31,6 +32,10 @@ func parseUpdate(flags *pflag.FlagSet, spec *api.ServiceSpec) error {
 			return err
 		}
 
+		if delayDuration < time.Duration(0) {
```

Comment:
This should be checked here:

https://github.com/docker/swarmkit/blob/master/manager/controlapi/service.go#L45